dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb

```diff
@@ -32,4 +35,63 @@ def index
 
     render json: submissions.values
   end
+
+  def report
+    enrollments = Plc::UserCourseEnrollment.where(plc_course_id: params[:course_id])
+    enrollment_submissions = Hash.new
+
+    scripts = Plc::Course.find(params[:course_id]).plc_course_units.map(&:script)
+    peer_reviewable_levels = ScriptLevel.where(script: scripts).select {|sl| sl.level.try(:peer_reviewable?)}.map(&:level).uniq
+
+    enrollments.each do |enrollment|
+      peer_review_submissions = Hash.new
+
+      UserLevel.where(user: enrollment.user, level: peer_reviewable_levels).each do |user_level|
+        status =
+          case user_level.best_result
+          when ActivityConstants::REVIEW_ACCEPTED_RESULT
+            'Accepted'
+          when ActivityConstants::REVIEW_REJECTED_RESULT
+            'Rejected'
+          when ActivityConstants::UNREVIEWED_SUBMISSION_RESULT
+            'Pending Review'
+          else
+            'Unsubmitted'
+          end
```

Comment:
This seems like a lot of logic in the controller. Is this case statement only needed in this controller action? Can we move it into the model or a serializer?

Reply:
Only used here. Might be able to move it to the model if there's enough peer review submission work to justify a model for peer review submissions (essentially a class that exists to unite peer review objects)

Reply:
Maybe then at least move the formatting into a private method so the controller action is a bit cleaner and easier to read?